main.cpp

```diff
@@ -1,52 +1,104 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <sstream>  //convert
+#include <cstring>  //strcmp
+#include <functional> //hash
 
-bool test_mode = true;
+namespace colors
```

Comment:
better to use enum classes ere instead of global variables inside the namespace

Reply:
I don't think that enum class is the best decision. I'll have to write function that converts enum variable into string, and calling it takes a bit more time than just global variables.